test/jdk/java/time/test/java/time/format/TestDateTimeFormatterBuilder.java

```diff
@@ -804,6 +808,78 @@ public void test_dayPeriodParseStrictNoTime() {
+        var dtf = builder
+                .appendValue(new TemporalField() {
+                                 @Override
+                                 public TemporalUnit getBaseUnit() {
+                                     return null;
+                                 }
+
+                                 @Override
+                                 public TemporalUnit getRangeUnit() {
+                                     return null;
+                                 }
+
+                                 @Override
+                                 public ValueRange range() {
+                                     return null;
+                                 }
+
+                                 @Override
+                                 public boolean isDateBased() {
+                                     return false;
+                                 }
+
+                                 @Override
+                                 public boolean isTimeBased() {
+                                     return false;
+                                 }
+
+                                 @Override
+                                 public boolean isSupportedBy(TemporalAccessor temporal) {
+                                     return false;
+                                 }
+
+                                 @Override
+                                 public ValueRange rangeRefinedBy(TemporalAccessor temporal) {
+                                     return null;
+                                 }
+
+                                 @Override
+                                 public long getFrom(TemporalAccessor temporal) {
+                                     return 0;
+                                 }
+
+                                 @Override
+                                 public <R extends Temporal> R adjustInto(R temporal, long newValue) {
+                                     return null;
+                                 }
+
+                                 @Override
+                                 public TemporalAccessor resolve(
+                                         Map<TemporalField, Long> fieldValues,
+                                         TemporalAccessor partialTemporal,
+                                         ResolverStyle resolverStyle) {
+                                     fieldValues.remove(this);
+                                     fieldValues.put(ChronoField.HOUR_OF_DAY, 6L);
+                                     return null;
+                                 }
+                             },
+                        1)
+                .appendPattern(" B")
+                .toFormatter()
+                .withLocale(Locale.US);
+        assertEquals((long)dtf.parse("0 in the morning").getLong(ChronoField.HOUR_OF_DAY), 6L);
+        try {
+            dtf.parse("0 at night");
+            throw new RuntimeException("DateTimeParseException should be thrown");
```

Comment:
Testng has `Assert.fail(message)` for this case.

Reply:
Fixed.